hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1178,16 +1216,22 @@ int SaraNcpClient::getAppFirmwareVersion() {
     // "L0.0.00.00.05.07,A.02.02" (demonstrator)
     // "L0.0.00.00.05.08,A.02.04" (maintenance)
     auto resp = parser_.sendCommand("ATI9");
-    int major = 0;
-    int minor = 0;
-    int r = CHECK_PARSER(resp.scanf("%*[^,],%*[A.]%d.%d", &major, &minor));
-    CHECK_TRUE(r == 2, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
-    CHECK_PARSER_OK(resp.readResult());
-    LOG(TRACE, "App firmware: %d", major * 100 + minor);
-    return major * 100 + minor;
+    int retMajor = 0, tempMajor = 0;
+    int retMinor = 0, tempMinor = 0;
+    int r = resp.scanf("%*[^,],%*[A.]%d.%d", &tempMajor, &tempMinor);
+    if (r == 2 && (resp.readResult() == AtResponse::OK)) {
+        retMajor = tempMajor;
+        retMinor = tempMinor;
+    }
+    LOG(TRACE, "App firmware: %d", retMajor * 100 + retMinor);
+    return retMajor * 100 + retMinor;
```

Comment:
Good call on the order. I don't mind adding your ver change as well.

Reply:
Regarding always reading the final result code "resp.readResult()", is that necessary to clear the parser of responses?

Reply:
IIRC, the destructor of `AtResponse` clears the context of the current AT command if the final response code wasn't read by the client code, so that the command channel can be reused for other commands.